src/lib.rs

```diff
@@ -266,29 +290,47 @@ impl Config {
         self
     }
 
-    fn is_flag_supported(&mut self, flag: &str) -> io::Result<bool> {
+    fn is_flag_supported(&mut self, flag: &str) -> Result<bool, Error> {
         let out_dir = self.get_out_dir();
         let src = out_dir.join("flag_check.c");
         if !self.check_file_created {
-            try!(write!(try!(fs::File::create(&src)), "int main(void) {{ return 0; }}"));
+            match fs::File::create(&src) {
+                Err(_) => return Err(Error {
+                    kind: ErrorKind::IOError,
+                    message: "Flag check failed to create temp file.".to_string(),
+                }),
+                Ok(mut f) => match write!(f, "int main(void) {{ return 0; }}") {
+                    Err(_) => return Err(Error {
+                        kind: ErrorKind::IOError,
+                        message: "Flag check failed to write to temp file.".to_string(),
+                    }),
+                    Ok(_) => (),
+                },
+            }
             self.check_file_created = true;
         }
 
         let obj = out_dir.join("flag_check");
-        let target = self.get_target();
+        let target = self.get_target()?;
         let mut cfg = Config::new();
         cfg.flag(flag)
            .target(&target)
            .opt_level(0)
            .host(&target)
            .debug(false)
            .cpp(self.cpp);
-        let compiler = cfg.get_compiler();
+        let compiler = cfg.get_compiler()?;
         let mut cmd = compiler.to_command();
         command_add_output_file(&mut cmd, &obj, target.contains("msvc"), false);
         cmd.arg(&src);
 
-        let output = try!(cmd.output());
+        let output = match cmd.output() {
+            Err(_) => return Err(Error {
+                kind: ErrorKind::ToolExecError,
+                message: "Flag check failed to obtain output.".to_string(),
```

Comment:
I think it'd be possible to cut down on the verbosity here by adding:

```rust
impl From<io::Error> for Error {
    // ...
}
```

and that way you can just use `?`!